### ISSUE TITLE
Export of files/attachments

### ISSUE THREAD

**Issue by ryan-adler** [zcutlip/pyonepassword#118]
When exporting from the Desktop client, it also exports files/attachments.  It doesn't appear that the pyonepassword includes these.  Please add this functionality.

**Comment by zcutlip** [comment]
Thank you for the GH issue!

Just to clarify, are you talking about mass-exporting many or all items? Or are you trying to dump specific items individually?

If you you're talking about some sort of mass export, that isn't really a supported feature of `pyonepassword`, but it does give you the tools you need to implement such a feature. There's a "dump_all" script under `scripts` I use for testing and diagnostics that should get you started. I may look to see if there's an elegant way to add file dumping to `dump_all.py`, but as I mentioned this isn't a supported feature at the moment.

If you're talking about dumping individual items from 1Password, the `OP.document_get()` method lets you download binary files stored in 1Password. You pass it the item name or UUID, and it returns a tuple of `(file_name, document_bytes)`.

Look at `examples/example-signin-get-document.py`.

Let me know if this helps, or if there's still something missing.

**Comment by ryan-adler** [comment]
Yes, I was referring to the dump_all essentially.  We would like to have similar functionality to the built-in 1PW export in the GUI, which exports everything including attachments.

**Comment by zcutlip** [comment]
I'm taking a look at this now. Here's something I noticed...

Files can be associated with items in one of two ways

**references**
In this case the item (e.g., a login item) has a reference to a document item. The document item is well understood, but to summarize, it has a "files" key (not part of any section or field) that contains list of file dictionaries. The dictionaries contain:
- file id
- file name (e.g., thumbnail.png)
- file size, in bytes

In the case of references, more than one item may reference a single file through its document item. In fact a single file may be referenced by unlimited items

Stated differently, and unlimited number of items may have an "attachment" which is actually the same file

> Note: a reference, while often a document item (with a file associated), may be any item type. So one login item, may reference another login item

**files**
It's also possible to add a file directly to an item (a login item or any other item) without a reference.

In this case the login (for example) item becomes like a "document item" in that it has "files" list as described above

-----

So, to download an item's attachments, two things need to be done:

1. Check if the item has any references, follow those references to see if they're documents, then retrieve the document bytes
    - `OP.document_get()` does most of this and serves as a model
2. Check if the item has a "files" list just like a document item would normally have. If it does, retrieve the file bytes for every entry in the list
